### PR TITLE
treefrog 1.13.0 (new formula)

### PR DESCRIPTION
**Pull request by skipbit**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/treefrog.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+class Treefrog < Formula
+  desc "High-speed C++ MVC Framework for Web Application"
+  homepage "http://www.treefrogframework.org/"
+  url "https://github.com/treefrogframework/treefrog-framework.git", :tag => "v1.13.0", :revision => "dad8d4bbb4ae0bfb480352fde3ba5dd1321ccff8"
```
Comment (MikeMcQuaid) [inline Formula/treefrog.rb]: Please split this onto multiple lines. Also, does this use submodules? If not, just use a tarball.

File: Formula/treefrog.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  head "https://github.com/treefrogframework/treefrog-framework.git", :branch => "master"
+
+  option "with-debug", "build with debug information"
+  option "with-gui", "build and link with QtGui module"
+  option "with-mongo", "build with MongoDB driver library"
+  option "with-mysql", "enable --with-mysql option for Qt build"
+  option "with-oci", "enable --with-oci option for Qt build"
+  option "with-postgresql", "enable --with-postgresql option for Qt build"
+
+  depends_on MinimumMacOSRequirement => :el_capitan
```
Comment (MikeMcQuaid) [inline Formula/treefrog.rb]: `depends_on :macos => :el_capitan`

File: Formula/treefrog.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  depends_on XcodeRequirement => [:build, :version => ">= 8.0"]
```
Comment (MikeMcQuaid) [inline Formula/treefrog.rb]: `depends_on :xcode => [:build, "8.0"]`

File: Formula/treefrog.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  qtopts = []
```
Comment (MikeMcQuaid) [inline Formula/treefrog.rb]: `qt5_build_options`

File: Formula/treefrog.rb [hunk 5]
```diff
@@ -0,0 +1,64 @@
+class Treefrog < Formula
+  desc "High-speed C++ MVC Framework for Web Application"
+  homepage "http://www.treefrogframework.org/"
+  url "https://github.com/treefrogframework/treefrog-framework.git", :tag => "v1.13.0", :revision => "dad8d4bbb4ae0bfb480352fde3ba5dd1321ccff8"
+  head "https://github.com/treefrogframework/treefrog-framework.git", :branch => "master"
+
+  option "with-debug", "build with debug information"
```
Comment (MikeMcQuaid) [inline Formula/treefrog.rb]: Omit this, we don't install symbols so it's unlikely this is useful.

File: Formula/treefrog.rb [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  option "with-gui", "build and link with QtGui module"
```
Comment (MikeMcQuaid) [inline Formula/treefrog.rb]: `with-qt5`

File: Formula/treefrog.rb [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  option "with-mongo", "build with MongoDB driver library"
```
Comment (MikeMcQuaid) [inline Formula/treefrog.rb]: `with-mongodb` or build by default if there's no dependency on it.

File: Formula/treefrog.rb [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+  def install
+    args = ["--prefix=#{prefix}"]
+
+    if build.with? "debug"
+      args << "--enable-debug"
+    end
+
+    if build.with? "gui"
+      args << "--enable-gui-mod"
```
Comment (MikeMcQuaid) [inline Formula/treefrog.rb]: `args << "--enable-gui-mod" if build.with? "qt5"` and similarly on the others.

File: Formula/treefrog.rb [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    end
+
+    cd "tools" do
+      system "make"
+      system "make", "install"
+    end
+  end
+
+  test do
+    cd testpath do
```
Comment (MikeMcQuaid) [inline Formula/treefrog.rb]: Can leave this; it's done by default.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by skipbit** [comment]
Thanks for review. All findings by @MikeMcQuaid are fixed.

File: Formula/treefrog.rb [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  sha256 "850744fef5254a8b2de1d949ba3490514ce1095e9a3e81da40aa8160129e14a1"
+  head "https://github.com/treefrogframework/treefrog-framework.git", :branch => "master"
+
+  option "with-mysql", "enable --with-mysql option for Qt build"
+  option "with-oci", "enable --with-oci option for Qt build"
+  option "with-postgresql", "enable --with-postgresql option for Qt build"
+  option "with-qt5", "build and link with QtGui module"
+
+  depends_on :macos => :el_capitan
+  depends_on :xcode => [:build, :version => ">= 8.0"]
```
Comment (MikeMcQuaid) [inline Formula/treefrog.rb]: `depends_on :xcode => [:build, "8.0"]` (as I said before)

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by skipbit** [comment]
fix mistake. Please check it.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks for your first contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**